js-api-doc/legacy/function.d.ts

```diff
@@ -79,9 +80,73 @@ export type LegacySyncFunction = (
  * APIs. Use [[CustomFunction]] with [[compile]], [[compileString]],
  * [[compileAsync]], and [[compileStringAsync]] instead.
  */
-export type LegacyAsyncFunction = (
-  this: LegacyPluginThis,
-  ...args: [...LegacyValue[], (result: LegacyValue) => void]
+export type LegacyAsyncFunction =
+  | ((this: LegacyPluginThis, done: (result: LegacyValue) => void) => void)
```

Comment:
shouldn't this use `LegacyAsyncFunctionDone` as well here ?

Reply:
Good catch!